packages/cli/src/ui/components/ProQuotaDialog.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type React from 'react';
+import { Box, Text } from 'ink';
+import { RadioButtonSelect } from './shared/RadioButtonSelect.js';
+
+interface ProQuotaDialogProps {
+  currentModel: string;
+  fallbackModel: string;
+  onChoice: (choice: 'auth' | 'continue') => void;
+}
+
+export function ProQuotaDialog({
+  currentModel,
+  fallbackModel,
+  onChoice,
+}: ProQuotaDialogProps): React.JSX.Element {
+  const items = [
+    {
+      label: 'Change auth (executes the /auth command)',
+      value: 'auth' as const,
+    },
+    {
+      label: `Continue with ${fallbackModel}`,
+      value: 'continue' as const,
+    },
+  ];
+
+  const handleSelect = (choice: 'auth' | 'continue') => {
+    onChoice(choice);
+  };
+
+  return (
+    <Box
+      borderStyle="round"
+      flexDirection="column"
+      paddingX={2}
+      paddingY={1}
```

Comment:
remove the Y padding to ensure this UI runs flicker free on more terminal sizes.